adding Kotlin guide

Anita-ihuman opened the pull request.

gememma commented on content/en/guides/how-to-debug-a-kotlin-microservice/index.md:
```diff
@@ -0,0 +1,258 @@
+## Introduction to debugging Kotlin microservices with mirrord
+With mirrord, we don’t have to think of building and releasing our applications for debugging. We can run our applications locally and mirrord will make sure to have your locally running process in the context of Kubernetes. Context mirroring for processes allows your process to run locally and consume the resources of a remote resource. 
+#### Workload to process context mirroring
+To achieve this, inputs from a Kubernetes workload (eg: a Pod) are mirrored to a locally running process. The process in question here today is a Kotlin process. Let’s see how we can mirror inputs for our locally running Kotlin application using mirrord and pipe these outputs back to Kubernetes. This will create a tighter feedback look effectively allowing you to debug faster without the downside ...
```
```suggestion
To achieve this, inputs from a Kubernetes workload (eg: a Pod) are mirrored to a locally running process. The process in question here today is a Kotlin process. Let’s see how we can mirror inputs for our locally running Kotlin application using mirrord and pipe these outputs back to Kubernetes. This will create a tighter feedback loop effectively allowing you to debug faster without the downsides of the common debugging techniques we discussed above.
```

gememma commented on content/en/guides/how-to-debug-a-kotlin-microservice/index.md:
```diff
@@ -0,0 +1,258 @@
+### Prerequisites
+Set up the Kubernetes cluster to test our application setup.
+
+1. Start an instance of a development cluster like minikube, k3d kind, etc. We are using minikube here.
```
```suggestion
1. Start an instance of a development cluster like minikube, k3d, kind, etc. We are using minikube here.
```

gememma commented on content/en/guides/how-to-debug-a-kotlin-microservice/index.md:
```diff
@@ -0,0 +1,258 @@
+We have our staging application deployed now. Let’s run the microservice with mirrord now. This will allow us to run the local Kotlin application in the context of Kubernetes without having to build and deploy it over and over again for testing.
+
+## Debug the Guestbook application with IntelliJ IDEA and the JetBrains mirrord plugin
+In this section of the guide, we are going to use the mirrord Intellij Plugin to help debug the Kotlin application. If you would like to see how we can do the same with the Kotlin Debugger in the CLI, go to this section of the guide.
```
> go to this section of the guide.

What section? I assume there should be a link here

gememma commented on content/en/guides/how-to-debug-a-kotlin-microservice/index.md:
```diff
@@ -0,0 +1,258 @@
+
+You will see a mirrord button and a dropdown menu in the top right corner of the screen.
+
+This dialogue box appears as mirrord is successfully installed. It contains the directions for how to use mirrord in the IntelliJ IDEA IDE. Let’s set up the mirrord configuration.
```
```suggestion
This dialog box appears as mirrord is successfully installed. It contains the directions for how to use mirrord in the IntelliJ IDEA IDE. Let’s set up the mirrord configuration.
```

gememma commented on content/en/guides/how-to-debug-a-kotlin-microservice/index.md:
```diff
@@ -0,0 +1,258 @@
+![alt text](<Screenshot 2025-03-14 at 4.04.09 PM.png>)
+
+
+If you would like to learn more about how the mirrord-agent in the above architecture works, go check out the reference here https://mirrord.dev/docs/reference/architecture/#mirrord-agent.
```
```suggestion
If you would like to learn more about how the mirrord-agent in the above architecture works, go check out the reference [here](https://mirrord.dev/docs/reference/architecture/#mirrord-agent).
```

gememma commented on content/en/guides/how-to-debug-a-kotlin-microservice/index.md:
```diff
@@ -0,0 +1,258 @@
+
+On the run above we can see that the application run fails because this local execution doesn’t have access to the services running inside the Kubernetes cluster we have created. 
+
+The microservice needs access to the “redis” service hosted on the cluster. To run the microservice with Kubernetes, we can use mirrord CLI tool.
```
```suggestion
The microservice needs access to the “redis” service hosted on the cluster. To run the microservice with Kubernetes, we can use the mirrord CLI tool.
```

gememma commented on content/en/guides/how-to-debug-a-kotlin-microservice/index.md:
```diff
@@ -0,0 +1,258 @@
+
+The microservice needs access to the “redis” service hosted on the cluster. To run the microservice with Kubernetes, we can use mirrord CLI tool.
+### 2. Install mirrord
+Let’s install mirrord CLI tool and run kotlin-guestbook with the required Kubernetes context. Follow the installation guide for mirrord here https://mirrord.dev/docs/overview/quick-start/#installation and run the below command.
```
```suggestion
Let’s install the mirrord CLI tool and run kotlin-guestbook with the required Kubernetes context. Follow the installation guide for mirrord [here](https://mirrord.dev/docs/overview/quick-start/#installation) and run the below command.
```

Anita-ihuman commented:
> marked some extra blank lines that need removing, also:
> 
> * all the numbered lists just show every number as "1."
>   ![Screenshot 2025-03-25 at 15 11 45](https://private-user-images.githubusercontent.com/58080601/426594801-8cd1b765-e59a-4935-aa5a-245086edb7a0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDI5ODMyMTEsIm5iZiI6MTc0Mjk4MjkxMSwicGF0aCI6Ii81ODA4MDYwMS80MjY1OTQ4MDEtOGNkMWI3NjUtZTU5YS00OTM1LWFhNWEtMjQ1MDg2ZWRiN2EwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzI2VDA5NTUxMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFiMDY4NzkwNjM0ZDY5YmU5ZjUzMDA1ODUyYzZmNGRhNjdhODM2NzMyMmQ2ZjA4NTVjMTI0NmNkMzQ2M2RiMjUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.CYSYfRrpVUs3ncJxDA2Jbw4dvl7zTxzeGwG42cOisWw)
> * all the images have the alt text "alt text" - they should have descriptions, even if its somethign short like "screenshot of console output"

addressed all comments

gememma commented on content/en/guides/how-to-debug-a-php-microservice/index.md:
````diff
@@ -0,0 +1,185 @@
+minikube start
+```
+
+1. Clone the repo with the sample PHP application.
````
```suggestion
2. Clone the repo with the sample PHP application.
```

gememma commented on content/en/guides/how-to-debug-a-php-microservice/index.md:
````diff
@@ -0,0 +1,185 @@
+```
+
+
+1. Deploy our application which will act as our staging environment.
````
```suggestion
3. Deploy our application which will act as our staging environment.
```

gememma commented on content/en/guides/how-to-debug-a-php-microservice/index.md:
```diff
@@ -0,0 +1,185 @@
+
+
+The minikube service command automatically sets up a port forwarding session to the specified service and opens it in the default web browser. With the tunnel to our microservice setup, our application architecture now looks like this. 
+![alt text](<Screenshot 2025-03-24 at 12.28.34 PM.png>)
```
```suggestion
![php application architecture](<Screenshot 2025-03-24 at 12.28.34 PM.png>)
```

gememma commented on content/en/guides/how-to-debug-a-php-microservice/index.md:
````diff
@@ -0,0 +1,185 @@
+🎉  Opening service default/guestbook in default browser...
+❗  Because you are using a Docker driver on darwin, the terminal needs to be open to run it.
+```
+![alt text](<Screenshot 2025-03-24 at 12.28.46 PM.png>)
````
```suggestion
![guestbook application webpage](<Screenshot 2025-03-24 at 12.28.46 PM.png>)
```

gememma commented on content/en/guides/how-to-debug-a-php-microservice/index.md:
```diff
@@ -0,0 +1,185 @@
+Now we have access to the Guestbook application on http://localhost:57485.
+Let’s access this URL from the browser.
+
+![alt text](<Screenshot 2025-03-24 at 12.29.07 PM.png>)
```
```suggestion
![console output](<Screenshot 2025-03-24 at 12.29.07 PM.png>)
```

gememma commented on content/en/guides/how-to-debug-a-php-microservice/index.md:
```diff
@@ -0,0 +1,185 @@
+
+The microservice needs access to the “redis” service hosted on the cluster. To run the PHP microservice with Kubernetes, we can use the mirrord CLI tool.
+
+1. Installing mirrord
```
```suggestion
2. Installing mirrord
```

gememma commented on content/en/guides/how-to-debug-a-php-microservice/index.md:
```diff
@@ -0,0 +1,185 @@
+
+Install the mirrord CLI tool and run Guestbook with the required Kubernetes context. Follow the installation guide for mirrord [here](https://mirrord.dev/docs/overview/quick-start/#installation) .
+
+1. Run the Guestbook application with php and mirrord in the CLI
```
```suggestion
3. Run the Guestbook application with php and mirrord in the CLI
```

gememma commented on content/en/guides/how-to-debug-a-php-microservice/index.md:
```diff
@@ -0,0 +1,185 @@
+
+On the run above we can see that the application run fails because this local execution doesn’t have access to the Redis instance running inside the Kubernetes cluster we have created. 
+
+![alt text](<Screenshot 2025-03-24 at 12.31.48 PM.png>)
```
```suggestion
![guestbook application failure](<Screenshot 2025-03-24 at 12.31.48 PM.png>)
```

gememma commented on content/en/guides/how-to-debug-a-php-microservice/index.md:
```diff
@@ -0,0 +1,185 @@
+
+The target impersonated Pod here is the Guestbook Pod. In this diagram, you can see how mirrord allows the user to use the mirrord-agent Pod as the execution context.
+
+![alt text](<Screenshot 2025-03-24 at 12.32.04 PM.png>)
```
```suggestion
![application diagram with mirrord](<Screenshot 2025-03-24 at 12.32.04 PM.png>)
```

gememma reviewed (approved):
🥳 👍
